src/conventions/h-group/urgent-actions.js

```diff
@@ -254,7 +256,7 @@ export function find_urgent_actions(game, play_clues, save_clues, fix_clues, sta
 			// Try to see if they have a playable card that connects directly through our hand
 			// Although this is only optimal for the next player, it is often a "good enough" action for future players.
 			const unlock_order = find_unlock(game, target);
-			if (unlock_order !== undefined) {
+			if (unlock_order !== undefined && (!finessed_card || finessed_card.order == unlock_order)) {
```

Comment:
Is there a reason we allow playing the finessed card into an unlock here and not when the next player is locked (line 219 filters it out)? It seems like the two situations should be similar.

Reply:
Updated the block in line 219 to allow this, it also makes sense that if they're locked we skip to looking at the next player rather than running the remainder of the block.